engine/src/main/java/org/hibernate/search/util/impl/ReflectionHelper.java

```diff
@@ -200,6 +200,36 @@ public static Object createInstance(Class<?> clazz, boolean checkForFactoryAnnot
 		return instance;
 	}
 
+	/**
+	 * Checks whether the specified type is a floating point type.
+	 *
+	 * @param type the type to check
+	 * @return {@code true} if the specified type is an floating point type or a wrapper thereof. {@code false} otherwise.
+	 */
+	public static boolean isFloatingPointType(Class<?> type) {
+		return float.class.equals( type )
```

Comment:
No, not at the moment. I was thinking a bit about it, but in order to use range faceting one would need to convert the `BigDecimal` to a double. This is potentially a conversion where precision is a problem. One would need a way to handle this. See also `BigDecimalNumericFieldBridge` in our docs which suggests a lossy numeric field bridge for `BigDecimal`. We can for sure think a bit more about it.